pulp_2to3_migration/app/plugin/rpm/pulp_2to3_models.py

```diff
@@ -141,44 +156,88 @@ class Pulp2Erratum(Pulp2to3Content):
     """
     Pulp 2to3 detail content model to store Pulp2 Errata content details.
 
-    Relations:
-        pulp2content(models.ManyToManyField): relation to the generic model. It needs to be
-                                              many-to-many because multiple Pulp2Content can
-                                              refer to the same detail model of errata.
-
     """
-
     # Required fields
     errata_id = models.TextField()
     updated = models.TextField()
     repo_id = models.TextField()
 
-    issued = models.TextField()
-    status = models.TextField()
-    description = models.TextField()
-    pushcount = models.TextField()
+    issued = models.TextField(null=True)
+    status = models.TextField(null=True)
+    description = models.TextField(null=True)
+    pushcount = models.TextField(null=True)
     references = JSONField()
     reboot_suggested = models.BooleanField()
     relogin_suggested = models.BooleanField()
     restart_suggested = models.BooleanField()
-    errata_from = models.TextField()
-    severity = models.TextField()
-    rights = models.TextField()
-    version = models.TextField()
-    release = models.TextField()
-    errata_type = models.TextField()
+    errata_from = models.TextField(null=True)
+    severity = models.TextField(null=True)
+    rights = models.TextField(null=True)
+    version = models.TextField(null=True)
+    release = models.TextField(null=True)
+    errata_type = models.TextField(null=True)
     pkglist = JSONField()
-    title = models.TextField()
-    solution = models.TextField()
-    summary = models.TextField()
+    title = models.TextField(null=True)
+    solution = models.TextField(null=True)
+    summary = models.TextField(null=True)
 
     pulp2_type = 'erratum'
     set_pulp2_repo = True
+    cached_repo_data = {}
 
     class Meta:
         unique_together = ('errata_id', 'repo_id')
         default_related_name = 'erratum_detail_model'
 
+    @classmethod
+    def get_repo_data(cls, pulp2_repo):
+        """
+        Get content data of a repository, NEVRA of packages and NSVCA of modules.
+
+        Args:
+            pulp2_repo(Pulp2Repository): a pre-migrated repo to collect data from
+
+        Returns:
+            dict: {'packages': list of NEVRA tuples of the packages which are in this repo,
+                   'modules': list of NSVCA tuples of the module which are in this repo}
+
+        """
+        if pulp2_repo.pk in cls.cached_repo_data:
+            return cls.cached_repo_data[pulp2_repo.pk]
+
+        repo_pkg_nevra = []
+        repo_module_nsvca = []
+
+        # gather info about available packages
+        package_pulp2_ids = Pulp2RepoContent.objects.filter(
+            pulp2_repository=pulp2_repo,
+            pulp2_content_type_id='rpm'
+        ).only('pulp2_unit_id').values_list('pulp2_unit_id', flat=True)
```

Comment:
can you add iterator() on this sort of queries?

Reply:
I'm not sure